packages/react-ui-core/src/Listing/ListingSchemas.js

```diff
@@ -0,0 +1,110 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+
+const ApartmentComplexSchema = ({ children, ...rest }) => (
+  <div itemType="https://schema.org/ApartmentComplex" itemScope {...rest}>
+    {children}
+  </div>
+)
+
+ApartmentComplexSchema.propTypes = {
+  children: PropTypes.node,
+}
+
+const AggregateRatingSchema = ({ score, count, reviews }) => {
+  if (!count && !reviews) {
+    return null
+  }
+
+  return (
+    <div
+      itemScope
+      itemProp="aggregateRating"
+      itemType="https://schema.org/AggregateRating"
+    >
+      <meta itemProp="worstRating" content="1.0" />
+      <meta itemProp="bestRating" content="5.0" />
+      <meta itemProp="ratingValue" content={score} />
+      {!!count && <meta itemProp="ratingCount" content={count} />}
+      {!!reviews && <meta itemProp="reviewCount" content={reviews} />}
+    </div>
+  )
+}
+
+AggregateRatingSchema.propTypes = {
+  score: PropTypes.number,
+  count: PropTypes.number,
+  reviews: PropTypes.number,
+}
+
+const AddressSchema = ({ addressLine1, city, state, zip }) => (
+  <span itemType="https://schema.org/PostalAddress" itemProp="address" itemScope>
+    <meta itemProp="streetAddress" content={addressLine1} />
+    <meta itemProp="addressLocality" content={city} />
+    <meta itemProp="addressRegion" content={state} />
+    <meta itemProp="postalCode" content={zip} />
+  </span>
+)
+
+AddressSchema.propTypes = {
+  addressLine1: PropTypes.string,
+  city: PropTypes.string,
+  state: PropTypes.string,
+  zip: PropTypes.string,
+}
+
+const GeoSchema = ({ latitude, longitude }) => {
+  if (!latitude || !longitude) {
+    return null
+  }
+
+  return (
+    <div itemProp="geo" itemScope itemType="https://schema.org/GeoCoordinates">
+      <meta itemProp="latitude" content={latitude} />
+      <meta itemProp="longitude" content={longitude} />
+    </div>
+  )
+}
+
+GeoSchema.propTypes = {
+  latitude: PropTypes.number,
+  longitude: PropTypes.number,
+}
+
+@themed(['PropertyLink'], { pure: true })
+class NameAndUrlSchema extends PureComponent {
+  render() {
+    const { url, theme, children } = this.props
+    return (
+      <div itemProp="name">
+        {url ?
+          <a
+            href={url}
+            itemProp="url"
+            target="_blank"
+            className={theme.PropertyLink}
+            onClick={event => event.preventDefault()}
+          >
```

Comment:
Does this cause problems if you try to nest another `a` tag as a child?

Reply:
Looks like it does! It probably won't come all that often though since the entire card is wrapped in an onClick so passing a link would be redundant anyway. The reason an anchor is added here is its better to tag structured data that's being rendered.

Reply:
Is there any way to test this?